internal/controller/changetransferpolicy_controller.go

```diff
@@ -473,3 +485,17 @@ func (r *ChangeTransferPolicyReconciler) mergePullRequests(ctx context.Context,
 
 	return nil
 }
+
+func TemplatePullRequest(prc *promoterv1alpha1.PullRequestConfiguration, ctp *promoterv1alpha1.ChangeTransferPolicy) (string, string, error) {
+	title, err := utils.RenderStringTemplate(prc.Template.Title, ctp)
```

Comment:
Maybe we should nest the CTP in case we want to later add other top-level config keys? @zachaller 

```suggestion
	title, err := utils.RenderStringTemplate(prc.Template.Title, map[string]any{"ctp": ctp})
```

Reply:
Yes, because we will want to add the `hydrator.metadata` as context I think eventually as well. But I would add that support in a follow up PR

Reply:
is it not already nested under `ctp`? Or are you think more than that.

Reply:
Good idea, it's updated